File: Student_Result_Processing_System/Services/ResultLoader.cs
using Models;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Services
{
    public class ResultLoader
    {
        public static List<Student> LoadStudents(string filepath, string errorLogPath)
        {
            List<Student> students = new List<Student>();

            if (!File.Exists(filepath))
            {
                Console.WriteLine($"Results file not found: {filepath}");
                return students;
            }

            Directory.CreateDirectory(Path.GetDirectoryName(errorLogPath));
            using(StreamWriter errorLog = new StreamWriter(errorLogPath, false))
            {
                foreach(string rawLine in File.ReadAllLines(filepath))
                {
                    string line = rawLine.Trim();

                    if (string.IsNullOrEmpty(line))
                    {
                        continue;
                    }

                    string[] parts = line.Split(',');

                    if(parts.Length != 5)
                    {
                        errorLog.WriteLine($"Skipped (missing one or more fields): {line}");
                        continue;
                    }

                    string name = parts[0];
                    string rollNo = parts[1];
                    string[] marksStr = { parts[2], parts[3], parts[4] };

                    List<int> marks = new List<int>();
                    bool conversionFailed = false;
                    string failedMark = "";

                    foreach(string m in marksStr)
                    {
                        if(int.TryParse(m, out int mValue))
                        {
                            marks.Add(mValue);
                        }
                        else
                        {
                            conversionFailed = true;
                            failedMark = m;
                            break;
                        }
                    }

                    if (conversionFailed)
                    {
                        errorLog.WriteLine($"Skipped (non-numeric mark '{failedMark}'): {line}");
                        continue;
                    }

                    Student student = new Student(name, rollNo, marks);
                    students.Add(student);
                }
            }
            return students;
        }
    }
}

File: Python/Student_Result_Processing_System/services/result_loader.py
import os
from models.student import Student

def load_students(filepath, error_log_path):
    students = []

    if not os.path.exists(filepath):
        print(f"Results file not found: {filepath}")
        return students

    os.makedirs(os.path.dirname(error_log_path), exist_ok=True)

    with open(filepath, "r") as f, open(error_log_path, "w") as error_log:
        for raw_line in f:
            line = raw_line.strip()

            if not line:
                continue  # skip blank lines silently, not an error

            parts = line.split(",")

            if len(parts) != 5:
                error_log.write(f"Skipped (missing one or more fields): {line}\n")
                continue

            name, roll_no, mark1_str, mark2_str, mark3_str = parts

            marks = []
            conversion_failed = False
            for mark_str in [mark1_str, mark2_str, mark3_str]:
                try:
                    marks.append(float(mark_str))
                except ValueError:
                    conversion_failed = True
                    break

            if conversion_failed:
                error_log.write(f"Skipped (non-numeric mark '{mark_str}'): {line}\n")
                continue

            student = Student(name, roll_no, marks)
            students.append(student)

    return students